crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,363 @@
+            ExpresscheckoutPaymentStatus::Authorized => Self::Authorized,
+            ExpresscheckoutPaymentStatus::PendingVbv => Self::AuthenticationPending,
+            ExpresscheckoutPaymentStatus::JuspayDeclined => Self::Failure,
+            ExpresscheckoutPaymentStatus::CaptureFailed => Self::CaptureFailed,
+            ExpresscheckoutPaymentStatus::CaptureInitiated => Self::CaptureInitiated,
+            ExpresscheckoutPaymentStatus::VoidInitiated => Self::VoidInitiated,
+            ExpresscheckoutPaymentStatus::VoidFailed => Self::VoidFailed,
+            ExpresscheckoutPaymentStatus::Voided => Self::Voided,
+            ExpresscheckoutPaymentStatus::PendingAuthentication => Self::AuthenticationPending,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ExpresscheckoutPaymentsResponse {
+    #[serde(default, deserialize_with = "deserialize_error_default")]
+    status: ExpresscheckoutPaymentStatus,
+    txn_uuid: Option<String>,
+    payment: Option<Authentication>,
+}
+
+fn deserialize_error_default<'de, D, ExpresscheckoutPaymentStatus>(
+    deserializer: D,
+) -> Result<ExpresscheckoutPaymentStatus, D::Error>
+where
+    ExpresscheckoutPaymentStatus: Default + Deserialize<'de>,
+    D: Deserializer<'de>,
+{
+    let opt = ExpresscheckoutPaymentStatus::deserialize(deserializer);
+    match opt {
+        Ok(v) => Ok(v),
+        _ => Ok(ExpresscheckoutPaymentStatus::default()),
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Authentication {
+    authentication: AuthenticationData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct AuthenticationData {
+    url: String,
+    method: Method,
+}
+
+impl<F, T>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            ExpresscheckoutPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            ExpresscheckoutPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::from(item.response.status),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: item.response.txn_uuid.map_or(
+                    types::ResponseId::NoResponseId,
+                    types::ResponseId::ConnectorTransactionId,
+                ),
+                redirection_data: item.response.payment.map(|r| services::RedirectForm {
+                    url: r.authentication.url,
+                    method: r.authentication.method,
+                    form_fields: HashMap::new(),
+                }),
+                redirect: true,
```

Comment:
As per my understanding this should be true only in case of redirection flow. Could you confirm that we will always redirection data for create payment?

Reply:
yes this will always be redirection